lib/dry/transaction/step.rb

```diff
@@ -56,7 +56,12 @@ def call(input)
       end
 
       def arity
-        operation.is_a?(Proc) ? operation.arity : operation.method(:call).arity
+        case operation
+        when Proc, Method
```

Comment:
I notice that the `arity` function was returning invalid data since we added the functionality of using `methods` as `steps`